R/assertions.R

```diff
@@ -1448,3 +1448,49 @@ assert_date_var <- function(dataset, var, dataset_name = NULL, var_name = NULL)
     ))
   }
 }
+
+#' Is an object a date or datetime vector?
+#'
+#' Check if an object/vector is a date or datetime variable without needing a dataset as input
+#'
+#' @param arg The function argument to be checked
+#'
+#' @param optional Is the checked parameter optional? If set to `FALSE`
+#' and `arg` is `NULL` then the function `assert_date_vector` exits early and throw and error.
+#'
+#' @return
+#' The function returns an error if `arg` is missing, or not a date or datetime variable
+#' but otherwise returns an invisible output.
+#'
+#' @export
+#'
+#' @author Sadchla Mascary
+#'
+#' @keywords assertion
+#'
+#' @family assertion
+#'
+#' @examples
+#' example_fun <- function(arg) {
+#'   assert_date_vector(arg)
+#' }
+#'
+#' example_fun(
+#'   as.Date("2022-01-30", tz = "UTC")
+#' )
+#' try(example_fun("1993-07-14"))
+assert_date_vector <- function(arg, optional = TRUE) {
+  assert_logical_scalar(optional)
+
+  if (optional && is.null(arg)) {
+    stop()
+  }
```

Comment:
This shouldn't throw an error but return just `return(invisible(arg))`. The rationale is that if a function has a parameter that is optional, i.e. its default value is `NULL` then the assertion should not do the checks below.

Reply:
update implemented